EmailAddressFinder/src/main/java/com/mycompany/emailaddressfinder/EmailAddressFinder.java

```diff
@@ -0,0 +1,53 @@
+package com.mycompany.emailaddressfinder;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+public class EmailAddressFinder {
+
+    private Set<String> emailAddresses = new HashSet();
+
+    public static void main(String[] args) throws Exception {
+        EmailAddressFinder finder = new EmailAddressFinder();
+        finder.findEmailsOnDomain("http://www.iana.org");
+    }
+
+    //ties it all together
+    public void findEmailsOnDomain(String domainName) throws IOException {
+        //assuming the protocol and domain name are valid
+        Document doc = Jsoup.connect(domainName).data("query", "Java").userAgent("Mozilla").cookie("auth", "token").timeout(3000).get();
+        getEmailAddresses(doc.toString());
+        Elements links = getPageLinks(doc);
+        for(Element link : links)
+        {
+            String linkAddress = link.attr("abs:href");
+            if(!linkAddress.contains("protocols"))
+                getEmailAddresses(Jsoup.connect(linkAddress).data("query", "Java").userAgent("Mozilla").cookie("auth", "token").timeout(3000).get().toString());
+        }
```

Comment:
Seems like this only checks 1 level deep on the page - as in, once you follow a link and parse out all the email addresses, it won't follow any more links on that second page.